MusicHub/antivirusProvider/service.py

```diff
@@ -0,0 +1,91 @@
+from time import sleep
+import json
+import requests
+
+from MusicHub.antivirusProvider.custom_exception import CustomAntiVirusException
+
+from ..config.settings import Common
+from .constants import (
+    AntivirusStatuses,
+    CHECK_PROGRESS_DELAY_SECONDS,
+    PROGRES_PERCENTAGE,
+)
+
+
+class AntivirusScan:
+    def __init__(self, filename):
+        self.file_url = Common.ANTIVIRUS_BASE_URL
+        self.post_headers = {
+            "Content-Type": "application/octet-stream",
+            "apikey": Common.ANTIVIRUS_API_KEY,
+            "filename": filename,
+        }
+        self.get_headers = {
+            "apikey": Common.ANTIVIRUS_API_KEY,
+            "x-file-metadata": "0",
+        }
+        self.bad_status = (
+            AntivirusStatuses.INFECTED,
+            AntivirusStatuses.SUSPICIOUS,
+            AntivirusStatuses.POTENTIALY_VULNERABLE_FILE,
+        )
+        self.failed_scan_status = (
+            AntivirusStatuses.FAILED_TO_SCAN,
+            AntivirusStatuses.ABORTED,
+            AntivirusStatuses.NOT_SCANNED,
+            AntivirusStatuses.CANCELED,
+        )
+        self.not_supported_status = AntivirusStatuses.FILETYPE_NOT_SUPPORTED
+
+    def scan_file_for_malicious_content(self, track):
+
+        response = self.sent_request(self.post_headers, "post", file=track)
+
+        response = self.wait_for_progress_finished(response)
+
+        self.check_result(response.get("scan_results").get("scan_all_result_i"))
+
+    def wait_for_progress_finished(self, response):
+        response = self.sent_request(
+            self.get_headers, "get", data_id=response.get("data_id")
+        )
```

Comment:
instead of 
```
        response = self.sent_request(
            self.get_headers, "get", data_id=response.get("data_id")
        )
```
might be 
```
wait_progress_request = {
   headers: self.get_headers(),
   data_id: response.get("data_id"),
   method: 'get'
}
  response = self.sent_request(**wait_progress_request)
```